### PR TITLE
specs::System as Processor

### PR DESCRIPTION
**Pull request by lschmierer**

### CODE REVIEW

**Comment by lschmierer** [comment]
The context is used to pass elapsed time to the processors.

File: src/ecs/src/lib.rs [hunk 1]
```diff
@@ -1,9 +1,10 @@
...
 extern crate specs;
 
-mod processor;
 mod sim;
 
-pub use specs::*;
-pub use self::processor::{Processor, ProcessorResult};
+use time::Duration;
+
+pub use specs::{World, Component, System as Processor, RunArg, Entity, EntityBuilder, Entities,
```
Comment (kvark) [inline src/ecs/src/lib.rs]: I wonder if it should just re-export `specs` directly instead.

File: src/ecs/src/sim.rs [hunk 2]
```diff
@@ -39,44 +38,35 @@ impl Simulation {
 
     /// Computes the next state of the world using the given processors.
     pub fn step(&mut self, dt: Duration) {
-        for p in self.procs.iter_mut() {
-            p.process(&mut self.planner, dt);
-        }
+        self.planner.dispatch(dt);
```
Comment (kvark) [inline src/ecs/src/sim.rs]: I think `dispatch` calls the wait for you

File: src/ecs/src/lib.rs (hunk 1 above)
Comment (csherratt) [inline src/ecs/src/lib.rs]: I have been thinking about doing this in a few places. The main reason being that users don't need to specify a version of your dependent library, which means less versioning problems when you upgrade a dependency.

Comment (kvark) [inline src/ecs/src/lib.rs]: if you just do `pub use specs;`, the users would neither need to specify a version

**Comment by kvark** [comment]
Thanks!
I think this is a good step forward, although not sure if we should really expose all the stuff inside `specs` instead of just exposing `specs` itself directly.

**Pull request merged**